### ISSUE TITLE
handle case where view is offscreen

### ISSUE THREAD

**Issue by MFlisar** [deano2390/MaterialShowcaseView#6]
If the view is wrapped in a scrollable container it would be necessary to make the view visible.

An easy solution would be to draw an arrow (instead of the circle) to the top, bottom, left or right of the screen and center the text on screen... (Just some idea that came to my mind)

Currently, the circle and the text will be positioned offscreen...

**Comment by deano2390** [comment]
Indeed, targeting views that are offscreen is a beast of a job. In my opinion this is beyond the scope of this library. However, making the showcase trigger once the target view is scrolled into the viewport might be a nice feature.

**Comment by MFlisar** [comment]
I inderstand, you are right. But at least, some acceptable solution should be integrated, as showing a mask with no content (no circle, no text) looks very bad.

That's why I suggested the easy solution... Indicate to the user that the text is meant for a view that currently is offscreen and display the text somewhere, where it at least is readable.

**Comment by ntoskrnl** [comment]
@MFlisar has a point! :+1:
A full-screen ,ask with no content looks bad!

**Comment by MFlisar** [comment]
Usecases are following for example:

* Having a scrollview and show the Showcase with a delay => the user may have been fast and have scrolled and the target is not visible
* Having a scrollview and having a help button in the actionbar => you anyway don't know what the current scroll position is

**Comment by ntoskrnl** [comment]
Sorry for my Kotlin

```kotlin
/**
 * Get partial visibility of this view (from 0.0 to 1.0).
 */
fun View.getPartialVisibility(): Float {
    if (isAttachedToWindow() && isShown()) {
        val windowRect = Rect()
        val drawingRect = Rect()
        val location = intArrayOf(0, 0)
        getWindowVisibleDisplayFrame(windowRect)
        getLocationInWindow(location)
        getDrawingRect(drawingRect)
        drawingRect.offset(location[0], location[1])
        val totalArea = drawingRect.width() * drawingRect.height().toFloat()
        if (drawingRect.intersect(windowRect)) {
            val visibleArea = drawingRect.width() * drawingRect.height().toFloat()
            return visibleArea / totalArea
        } else return 0.0f
    } else {
        return 0.0f
    }
}
```

Maybe before showing the view we could make this simple check? It will not work for all cases, but for simple (and most common cases) it will.

**Comment by nologinatgit** [comment]
Yeah, I agree with @MFlisar , without the proper positioning for the offscreen views, this library is not usable. It should create at least a _scollToContent _method, to make the view (hence the guide) visible. Too bad, I have to look for another solution.
